lib/presentation/auth/change_password/confirm/bloc/change_password_confirm_cubit.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_template/data/auth/model/auth_data_error.dart';
+import 'package:flutter_template/data/auth/repository/auth_repository.dart';
+import 'package:flutter_template/data/auth/service/remote/model/forgot_password_confirm/forgot_password_confirm_request_model.dart';
+import 'package:flutter_template/presentation/auth/change_password/confirm/bloc/change_password_confirm_error.dart';
+import 'package:flutter_template/presentation/auth/change_password/confirm/bloc/change_password_confirm_state.dart';
+import 'package:flutter_template/util/extensions/string_extension.dart';
+import 'package:flutter_template/util/mixins/resettable_bloc_mixin.dart';
+
+class ChangePasswordConfirmCubit extends Cubit<ChangePasswordConfirmState>
+    with ResettableBlocMixin {
+  final AuthRepository _authRepository;
+
+  ChangePasswordConfirmCubit(this._authRepository)
+      : super(
+          const ChangePasswordConfirmStateInitial(
+            email: "",
+            password: "",
+            confirmPassword: "",
+            code: "",
+          ),
+        );
+
+  @override
+  void resetState() {
+    emit(
+      ChangePasswordConfirmStateInitial(
+        email: state.email,
+        password: state.password,
+        confirmPassword: state.confirmPassword,
+        code: state.code,
+      ),
+    );
+  }
+
+  @override
+  void onChange(Change<ChangePasswordConfirmState> change) {
+    super.onChange(change);
+
+    // TODO: Add Analytics and logging here
+  }
+
+  void setPassword(String password) {
+    emit(
+      state.copyWith(password: password),
+    );
+  }
+
+  void setConfirmPassword(String confirmPassword) {
+    emit(
+      state.copyWith(confirmPassword: confirmPassword),
+    );
+  }
+
+  void setCode(String code) {
+    emit(
+      state.copyWith(code: code),
+    );
+  }
+
+  void retrieveEmail(String? email) {
+    if (!email.isNullOrEmpty) {
+      emit(
+        state.copyWith(email: email),
+      );
+    }
+  }
+
+  Future<void> changePassword({
+    required String email,
+    required String code,
+    required String password,
+  }) async {
```

Comment:
`email` is a required parameter, but then it's taken from the state?
Same as with others, unclear whether the strategy is to keep values in the state, or pass to the function as parameter. We should pick one or the other